aws/lambda/pytorch-auto-revert/pytorch_auto_revert/testers/hud.py

```diff
@@ -1,94 +1,76 @@
+import json
 import logging
-from typing import Iterable, Optional
+from typing import Any, Mapping, Optional, Union
 
-from ..hud_renderer import build_grid_model, render_html
-from ..signal_extraction import SignalExtractor
+from ..hud_renderer import render_html_from_state
+from ..signal_extraction_datasource import SignalExtractionDatasource
 
 
-def run_hud(
-    workflows: Iterable[str],
-    *,
-    hours: int = 24,
-    repo_full_name: str = "pytorch/pytorch",
-    out: str = "hud.html",
-    ignore_newer_than: Optional[str] = None,
-) -> str:
-    """
-    Extracts signals for the given workflows, optionally truncates commit history
-    to ignore commits newer than a specific SHA, builds a HUD model, and writes
-    an HTML report to `out`.
+RunStatePayload = Union[str, Mapping[str, Any]]
+
 
-    Returns the output filepath.
-    """
+def _ensure_state_dict(state: RunStatePayload) -> Mapping[str, Any]:
```

Comment:
please add a strict jsonschema validation here

otherwise errors will be random, criptic and hard to understand if someone provides accidentally the wrong json file.